client/components/domains/connect-domain-step/index.jsx

```diff
@@ -94,16 +115,28 @@ function ConnectDomainStep( { domain, selectedSite, initialSetupInfo, initialSte
 				.catch( ( error ) => setDomainSetupInfoError( { error } ) )
 				.finally( () => setLoadingDomainSetupInfo( false ) );
 		} )();
-	}, [ domain, domainSetupInfo, initialSetupInfo, loadingDomainSetupInfo, selectedSite.ID ] );
+
+		return () => {
+			setDomainSetupInfo( {} );
+			setLoadingDomainSetupInfo( false );
+		};
```

Comment:
I wonder if we need to clean these two state values because they're reset when the component is unmounted, aren't they?